worksheets/espresso-testing/index.md

```diff
@@ -0,0 +1,240 @@
+        exclude group: 'com.android.support', module: 'support-annotations'
+    })
+}
+```
+
+Press the *Sync Now* button in the bar which has appeared along the top. If the bar disappears after syncing, then everything is set up correctly! If the bar shows and error message and a *Try Again* button then something has gone wrong. Take a look at the error at the bottom of the screen to see if you can fix it, or ask your coach for help.
+
+## Writing our first test
+Now we have everything set up; we can write our first test!
+
+In the project navigator on the left panel of Android Studio, navigate through `app -> java -> <package name> (androidTest)` where `<package name>` is the name you used when you set up the project. Right click on the `(androidTest)` line and choose `New -> Java Class`. In the window which pops up, give it a name of `CookieClickerTest` and then press `OK`. This will create and open a file which looks like this:
+
+```java
+package io.codebar.cookieclicker;
+
+public class CookieClickerTest {
+}
+```
+
+Above the `public class` line, add in this line:
+
+```java
+@RunWith(AndroidJUnit4.class)
+```
+
+Remember to use the autocomplete as much as possible as it will also add in some `import` lines above for you. This line tells Android how to run the tests we are about to write.
+
+Inside the `class` add the following line:
+
+```java
+@Rule
+public ActivityTestRule<MainActivity> activityTestRule = new ActivityTestRule<>(MainActivity.class);
+```
+
+This is a "test rule" and test Espresso which activity (part of our app) we want to test. We have chosen the `MainActivity`. This rule will make Espresso open this activity before running our test so we can see the cookie clicker and test it is working.
+
+We now need to actually write some test code. Inside the `class`, create a new method called `totalStartsAtZero`. It should look like this:
+
+```java
+@Test
+public void totalStartsAtZero() throws Exception {
+
+}
+```
+
+We will be writing one of these methods for each test that we want to perform. As you can guess from the name, we are first going to check that the counter starts off at zero.
+
+Inside the method, write the following code:
+
+```java
+onView(withId(R.id.lblTotal))
+    .check(matches(withText("0")));
+```
+
+So what are we doing here? The first line (`onView`) is finding the view which matches the requirements we have given to it. We have asked for the view with the id of `lblTotal` using `withId(R.id.lblTotal)`. Remember that this is the id we gave to our `TextView` in the first tutorial in the `activity_main.xml` layout file. The second line is checking that the view matches some conditions. We are checking that the view has the text `"0"` using `withText("0")`.
+
+This pattern of getting the view with a set of requirements and then checking something with it forms the basis of how Expresso tests work.
+
+The completed test file should now look like this:
```

Comment:
again, maybe consider using a screenshot? just so people don't copy paste it all and screw up package names etc?